code/game/objects/structures/mineral_doors.dm

```diff
@@ -83,26 +83,22 @@
 		icon_state = "[base_icon_state][smoothing_flags ? "-[smoothing_junction]" : ""]"
 
 
-/obj/structure/mineral_door/attackby(obj/item/W, mob/living/user)
+/obj/structure/mineral_door/attackby(obj/item/attacking_item, mob/living/user)
 	. = ..()
 	if(.)
-		return
+		return TRUE
 	if(QDELETED(src))
 		return
+	if(istype(attacking_item, /obj/item/tool/pickaxe/plasmacutter) && !user.do_actions) //Special plasmacutter interaction
+		var/obj/item/tool/pickaxe/plasmacutter/pcutter = attacking_item
+		if(pcutter.start_cut(user, name, src, PLASMACUTTER_BASE_COST * PLASMACUTTER_VLOW_MOD))
+			pcutter.cut_apart(user, name, src, PLASMACUTTER_BASE_COST * PLASMACUTTER_VLOW_MOD) //Minimal energy cost.
+			user.changeNext_move(attacking_item.attack_speed)
+			user.do_attack_animation(src, used_item = attacking_item)
+			take_damage(src.max_integrity, BURN, FIRE, armour_penetration = 100)
+		return
 
-	var/multiplier = 1
-	if(istype(W, /obj/item/tool/pickaxe/plasmacutter) && !user.do_actions)
-		var/obj/item/tool/pickaxe/plasmacutter/P = W
-		if(P.start_cut(user, src.name, src, PLASMACUTTER_BASE_COST * PLASMACUTTER_VLOW_MOD))
-			if(istype(src, /obj/structure/mineral_door/resin))
-				multiplier += PLASMACUTTER_RESIN_MULTIPLIER //Plasma cutters are particularly good at destroying resin structures.
-			else
-				multiplier += PLASMACUTTER_RESIN_MULTIPLIER * 0.5
-			P.cut_apart(user, src.name, src, PLASMACUTTER_BASE_COST * PLASMACUTTER_VLOW_MOD) //Minimal energy cost.
-	if(W.damtype == BURN && istype(src, /obj/structure/mineral_door/resin)) //Burn damage deals extra vs resin structures (mostly welders).
-		multiplier += 1 //generally means we do double damage to resin doors
-
-	take_damage(max(0, W.force * multiplier - W.force), W.damtype, MELEE)
+	attacking_item.attack_obj(src, user)
```

Comment:
This makes attack_obj run twice if it returns false. Look at how obj/alien does it.